src/Compilers/CSharp/Portable/Symbols/Source/SourceTypeParameterSymbol.cs

```diff
@@ -864,6 +889,15 @@ public override bool HasValueTypeConstraint
             }
         }
 
+        public override bool IsValueTypeFromConstraintTypes
+        {
+            get
+            {
+                var typeParameter = this.OverriddenTypeParameter;
+                return ((object)typeParameter != null) && (typeParameter.IsValueTypeFromConstraintTypes || CalculateIsValueTypeFromConstraintTypes(ConstraintTypesNoUseSiteDiagnostics));
```

Comment:
> || CalculateIsValueTypeFromConstraintTypes(ConstraintTypesNoUseSiteDiagnostics) [](start = 103, length = 80)

nit: just curious, why do we need this part of the check? I didn't get why `typeParameter.IsValueTypeFromConstraintTypes` isn't sufficient.

Reply:
> just curious, why do we need this part of the check? I didn't get why typeParameter.IsValueTypeFromConstraintTypes isn't sufficient.

Overriding often involves substitution, constraint types that were not known to be a value type can be substituted with value types.

---
In reply to: [520174331](https://github.com/dotnet/roslyn/pull/49227#discussion_r520174331) [](ancestors = 520174331)